### PR TITLE
[NCL-4012] implemented refershing access tokens for user

### PR DESCRIPTION
**Pull request by michalovjan**

### CODE REVIEW

**Comment by thescouser89** [comment]
test this please

File: pnc_cli/user_config.py [hunk 1]
```diff
@@ -70,16 +71,23 @@ def __setstate__(self, state):
         else:
             self.keycloak_config = saved_kc_config
 
+        if not newtoken and 8640000 >= utils.current_time_millis() - self.token_time > 360000:
```
Comment (janinko) [inline pnc_cli/user_config.py]: you can't use one one variable for remembering both login and refresh time

File: pnc_cli/user_config.py [hunk 2]
```diff
@@ -16,8 +16,7 @@
 import keycloak_config as kc
 import pnc_server_config as psc
 
-
-
+# make sure that input behaves as expected
```
[the remark this answers is not in the document]
Comment (michalovjan) [reply pnc_cli/user_config.py]: pycharm autoPEP-ed

[the remark this answers is not in the document]
Comment (janinko) [reply pnc_cli/user_config.py]: The comment is there twice

File: pnc_cli/user_config.py [hunk 3]
```diff
@@ -70,25 +71,34 @@ def __setstate__(self, state):
...
+                self.access_token = self.retrieve_keycloak_token()
             else:
                 # enter password to get new token, but only if the user has not entered a password in pnc-cli.conf
                 password = self.load_password_from_config(config)
                 if password:
                     self.password = password
                 else:
                     self.password = self.input_password()
-                self.token = self.retrieve_keycloak_token()
+                self.access_token = self.retrieve_keycloak_token()
```
Comment (janinko) [inline pnc_cli/user_config.py]: When you are setting the refresh token inside the method, you can set there the acess_token as well, no?

File: pnc_cli/user_config.py [hunk 4]
```diff
@@ -70,25 +71,34 @@ def __setstate__(self, state):
         else:
             self.keycloak_config = saved_kc_config
 
+
         # if more than a day has passed since we saved the token, and keycloak server configuration is not modified,
         # get a new one
-        if not newtoken and utils.current_time_millis() - self.token_time > 86400000:
+        if not newtoken and utils.current_time_millis() - self.refresh_token_time > 8640000:
```
Comment (janinko) [inline pnc_cli/user_config.py]: Even though it probably doesn't help much, I would change `>` to `>=`

File: pnc_cli/user_config.py (hunk 3 above)
Comment (michalovjan) [reply pnc_cli/user_config.py]: That was the method's original functionality (that it returns access_token). Same thing is happening in refresh_access_token(self) method. So, this change should be made, I'd rather do it in both retrieve and refresh methods for the sake of consistency.

File: pnc_cli/user_config.py (hunk 4 above)
Comment (michalovjan) [reply pnc_cli/user_config.py]: Sure thing.

**Comment by michalovjan** [comment]
@thescouser89 Hi Dustin, @janinko told me to ask you to review my code for this issue because he is on PTO. Just to be sure :).

**Comment by thescouser89** [comment]
@michalovjan  Noted! Will review soon

**Comment by thescouser89** [comment]
I'm not very familiar with Keycloak unfortunately but the logic looks sane. I'll approve the request! :)

**Pull request merged**